src/Misc.ino

```diff
@@ -158,28 +158,33 @@ void deepSleep(int delay)
     }
   }
   saveUserVarToRTC();
-  deepSleepStart(delay); // Call deepSleepStart function after these checks
+  deepSleepStart(dsdelay); // Call deepSleepStart function after these checks
 }
 
-void deepSleepStart(int delay)
+void deepSleepStart(int dsdelay)
 {
   // separate function that is called from above function or directly from rules, usign deepSleep as a one-shot
   String event = F("System#Sleep");
   rulesProcessing(event);
 
-
   RTC.deepSleepState = 1;
   saveToRTC();
 
-  if (delay > 4294 || delay < 0)
-    delay = 4294;   //max sleep time ~1.2h
-
   addLog(LOG_LEVEL_INFO, F("SLEEP: Powering down to deepsleep..."));
+  delay(100); // give the node time to send above log message before going to sleep
   #if defined(ESP8266)
-    ESP.deepSleep((uint32_t)delay * 1000000, WAKE_RF_DEFAULT);
+    #if defined(CORE_2_5_0)
+      if (dsdelay > ESP.deepSleepMax() || dsdelay < 0)
+        dsdelay = ESP.deepSleepMax();   //max sleep time
+        ESP.deepSleepInstant((uint32_t)dsdelay * 1000000, WAKE_RF_DEFAULT);
+    #else
```

Comment:
The function returns:
 uint64_t deepSleepMax()

And the function is called with a value in seconds and the instant function has to be called in usec.
So the value has to be uint64_t at least and you must compare that million times larger value with the `deepSleepMax()`.

Reply:
Ok, so I hope I got the types and casts correct this time ;)

I made sure the delay stays <= INT_MAX so that the parameter storage of the deep sleep time `Settings.Delay` does not have to be changed.

Reply:
I'm not really sure about your fix, but will have a look at it.

Reply:
great! Me neither, so it's good to have at least a second pair of eyes going through it ;)
thanks!!